addons/point_of_sale/tests/test_point_of_sale_flow.py

```diff
@@ -1341,6 +1356,12 @@ def test_sale_order_postponed_invoicing(self):
             statement_receivable_line = statement.line_ids.filtered(lambda line: line.account_id == self.company_data['default_account_receivable'] and line.name == pos_order.session_id.name)  # Because the payment method use the default receivable
             self.assertEqual(closing_entry_receivable_line.matching_number, statement_receivable_line.matching_number)
 
+            # Check 3: Tax tags are correctly set up for tax reports
+            invoice_product_line = invoice.line_ids.filtered(lambda line: line.display_type == 'product')
+            self.assertEqual(invoice_product_line.tax_tag_ids.name, 'invoice')
+            misc_product_line = misc_reversal_entry.line_ids.filtered(lambda line: line.display_type == 'product')
+            self.assertEqual(misc_product_line.tax_tag_ids.name, 'refund')
```

Comment:
Functionally speaking, do we really want to use the refund tag on the reversal ? We revert it to redo it on an invoice, right, so it's not really a refund :thinking: IMO it would make sense to have the same repartition, hence same tag, but in negative (so, inverting the sign of the chosen tag(s)). But maybe I'm wrong, and I'm not a POS expert ; did you discuss that with a PO?

Reply:
> We revert it to redo it on an invoice, right, so it's not really a refund 
Conceptually seems the same of adding a credit note selecting 'Full refund and new draft invoice', so it kinda make sense to me, but I'm not an accountant :D
I'll check with a PO to be sure

Reply:
Not really if I understand it right. 
In the "full refund and new draft invoice" flow, the original invoice, its refund and the new invoice can have three different accounting dates. Therefore, the refund has to be considered as such (a refund). Since, assuming no other entry elsewhere, it could result in a negative grid 03 in the month/quarter of the out_refund. >> Belgian Tax Report would be rejected upon submission to Intervat. 

While in the PoS flow, the entry that cancels the initial booking and the invoice addressed to a specific customer will necessarily happen the same day and share the same accounting date. 

Or am I missing something?